apps/web/pages/api/og.tsx

```diff
@@ -0,0 +1,291 @@
+).then((res) => res.arrayBuffer());
+
+type UrlSet = {
+  small: string | null;
+  medium: string | null;
+  large: string | null;
+};
+
+export default async function handler(request: NextRequest) {
+  const { searchParams } = request.nextUrl;
+
+  const postId = searchParams.get('postId') ?? '';
+  let postImageUrl =
+    'https://assets.gallery.so/https%3A%2F%2Fstorage.googleapis.com%2Fprod-token-content%2F4-292cd-KT1EfsNuqwLAWDd3o4pvfUx1CAh5GMdTrRvr-image?auto=compress%2Cformat&fit=max&glryts=1705844681&w=1024&s=9076d07060aeb7ac31297a0381bd0ed3';
+ // const caption = 'grain:street #63 by nekropunk';
+
+  const queryResponse = await fetchWithJustQueryText({
+    queryText: postIdQuery,
+    variables: { postId: postId },
+  });
+
+  if (!postId || !queryResponse?.data?.post) {
+    return new ImageResponse(<>Visit gallery.so;</>, {
+      width: 1200,
+      height: 630,
+    });
+  }
+
+  const post = queryResponse.data.post;
+  const author = post.author;
+  const firstLetter = author?.username?.substring(0, 1).toUpperCase() ?? '';
+
+  let profileImageUrl = null;
+  const { token: profileToken, profileImage } = post?.author?.profileImage ?? {};
+
+  if (profileImage && profileImage.previewURLs?.medium) {
+    profileImageUrl = profileImage.previewURLs.medium;
+  }
+
+  const media = profileToken?.definition?.media;
+
+  let previewUrls: UrlSet | null = null;
+
+  if (!profileToken) {
+    return null;
+  }
+
+  if (
+    media &&
+    'previewURLs' in media &&
+    media.previewURLs &&
+    (media.previewURLs.small || media.previewURLs.medium || media.previewURLs.large)
+  ) {
+    previewUrls = media.previewURLs;
+  } else if ('fallbackMedia' in media) {
+    if (media.fallbackMedia?.mediaURL) {
+      previewUrls = {
+        small: media.fallbackMedia.mediaURL,
+        medium: media.fallbackMedia.mediaURL,
+        large: media.fallbackMedia.mediaURL,
+      };
+    }
+  }
+
+  profileImageUrl = previewUrls?.small;
+
+  const postToken = post.tokens?.[0];
+  if (!postToken) {
+    return null;
+  }
+
+  const postMedia = postToken?.definition?.media;
+  let postPreviewUrls: UrlSet | null = null;
+
+  if (
+    postMedia &&
+    'previewURLs' in postMedia &&
+    postMedia.previewURLs &&
+    (postMedia.previewURLs.small || postMedia.previewURLs.medium || postMedia.previewURLs.large)
+  ) {
+    postPreviewUrls = postMedia.previewURLs;
+  } else if ('fallbackMedia' in media) {
+    if (media.fallbackMedia?.mediaURL) {
+      postPreviewUrls = {
+        small: media.fallbackMedia.mediaURL,
+        medium: media.fallbackMedia.mediaURL,
+        large: media.fallbackMedia.mediaURL,
+      };
+    }
+  }
+
+  postImageUrl = postPreviewUrls?.large ?? '';
+
+  const ABCDiatypeRegularFontData = await ABCDiatypeRegular;
+  const ABCDiatypeBoldFontData = await ABCDiatypeBold;
+  const alpinaLightFontData = await alpinaLight;
+
+  return new ImageResponse(
```

Comment:
lower priority - let's wrap this whole thing in a try/catch and in the catch return a new `ImageResponse` featuring the gallery logo as an SVG (this should be in our asset folders in the gallery repo)

we should also throw an error somewhere in here if the media isn't available. if so, `catch` will handle it and render the backup